### PR TITLE
:sparkles: Adding ability to set analysis modes per provider init config

### PR DESCRIPTION
**Pull request by shawn-hurley**
1. Adding a CLI option to override those options

### CODE REVIEW

File: provider/internal/builtin/provider.go [hunk 1]
```diff
@@ -120,6 +120,9 @@ func (p *builtinProvider) ProviderInit(context.Context) error {
 
 // We don't need to init anything
 func (p *builtinProvider) Init(ctx context.Context, log logr.Logger, config provider.InitConfig) (provider.ServiceClient, error) {
+	if config.AnalysisMode != provider.AnalysisMode("") {
+		p.log.V(5).Info("skipping analsyis mode setting for builtin")
```
Comment (pranavgaikwad) [inline provider/internal/builtin/provider.go:124]: nit: /analsyis/analysis/

File: provider/internal/java/provider.go [hunk 2]
```diff
@@ -173,6 +173,13 @@ func (p *javaProvider) ProviderInit(ctx context.Context) error {
 }
 
 func (p *javaProvider) Init(ctx context.Context, log logr.Logger, config provider.InitConfig) (provider.ServiceClient, error) {
+	//By default if nothing is set for analysis mode, in the config, we should default to full for external providers
+	var a provider.AnalysisMode = provider.AnalysisMode(config.AnalysisMode)
+	if a == provider.AnalysisMode("") {
+		a = provider.FullAnalysisMode
+	} else if !(a == provider.FullAnalysisMode || a == provider.ParitalAnalysisMode || a == provider.SourceOnlyAnalysisMode) {
```
Comment (pranavgaikwad) [inline provider/internal/java/provider.go:180]: Probably move this check to a common helper function?

File: provider/server.go [hunk 3]
```diff
@@ -86,10 +86,19 @@ func (s *server) Capabilities(ctx context.Context, _ *emptypb.Empty) (*libgrpc.C
 }
 
 func (s *server) Init(ctx context.Context, config *libgrpc.Config) (*libgrpc.InitResponse, error) {
+	//By default if nothing is set for analysis mode, in the config, we should default to full for external providers
+	var a AnalysisMode = AnalysisMode(config.AnalysisMode)
+	if a == AnalysisMode("") {
+		a = FullAnalysisMode
+	} else if !(a == FullAnalysisMode || a == ParitalAnalysisMode || a == SourceOnlyAnalysisMode) {
```
Comment (pranavgaikwad) [inline provider/server.go:93]: Helper function?

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
looks good overall, just some minor changes

File: provider/internal/java/provider.go (hunk 2 above)
Comment (shawn-hurley) [reply provider/internal/java/provider.go:180]: I don't think it is necessary, soon the only place this code will live is in the provider-server as java will move to an external provider as well.

File: provider/server.go (hunk 3 above)
Comment (shawn-hurley) [reply provider/server.go:93]: See previous comment.

**Review by pranavgaikwad — APPROVED** [review]

**Pull request merged**